### PR TITLE
Adding in new function

### PR DESCRIPTION
**Pull request by ajsockol**
- [ ] Closes #xxxx
- [x] Tests added
- [x] Documentation reflects changes
- [x] PEP8 Standards or use of linter
- [x] Xarray Dataset or DataArray variable naming follows 'ds' or 'da' naming

### CODE REVIEW

File: act/utils/data_utils.py [hunk 1]
```diff
@@ -1410,3 +1410,112 @@ def calculate_percentages(ds, fields, time=None, time_slice=None, threshold=None
         percentages[i] = j
     ds_percent.close()
     return percentages
+
+
+def convert_2d_to_1d(ds_object, parse=None, variables=None, keep_name_if_one=False,
```
Comment (zssherman) [inline act/utils/data_utils.py:1415]: @ajsockol Can we rename ds_object to ds in this new code to be consistent with other functions in ACT.

File: act/utils/data_utils.py [hunk 2]
```diff
@@ -1410,3 +1410,112 @@ def calculate_percentages(ds, fields, time=None, time_slice=None, threshold=None
...
+                     use_dim_value_in_name=False, dim_labels=None):
+    """
+    Function to convert a single 2D variable into multiple 1D
+    variables using the second dimension in the new variable name.
+
+    ...
```
Comment (zssherman) [inline act/utils/data_utils.py:1421]: Can we remove the whitespace and 3 . line and leave one space between intro sentence and Parameter header

File: act/utils/data_utils.py [hunk 3]
```diff
@@ -1410,3 +1410,112 @@ def calculate_percentages(ds, fields, time=None, time_slice=None, threshold=None
...
+        Option to not modify the variable name if the coordinate dimension
+        has only one value. Essentially converting a 2D (i.e. (100,1)
+        variable into a 1D variable (i.e. (100)).
+    use_dim_value_in_name: boolean
+        Option to use value from the coordinate dimension in new variable
+        name instead of indexing number. Will use the value prepended
+        to the units of the dimension.
+    dim_labels: str or list of str
+        Allows for use of custom label to append to end of variable names
+
```
Comment (zssherman) [inline act/utils/data_utils.py:1445]: Removal of one space

File: act/utils/data_utils.py [hunk 4]
```diff
@@ -1410,3 +1410,112 @@ def calculate_percentages(ds, fields, time=None, time_slice=None, threshold=None
...
+
+    Returns
+    -------
+        A new object copied from input object with the multi-dimensional
+        variable split into multiple single-dimensional variables.
+
+
```
Comment (zssherman) [inline act/utils/data_utils.py:1452]: Remove 1 space as well her
e

File: act/utils/data_utils.py [hunk 5]
```diff
@@ -1410,3 +1410,112 @@ def calculate_percentages(ds, fields, time=None, time_slice=None, threshold=None
...
+    # Now use the parse_dim name to parse the variable and return new object.
+    >>> new_ds_object = convert_2d_to_1d(ds_object, parse=parse_dim)
+
+    """
+
+    # If no parse dimension name given assume it is the one not equal to 'time'
+    if parse is None:
+        parse = (list(set(list(ds_object.dims)) - set(['time'])))[0]
+
+    new_ds_object = ds_object.copy()
```
Comment (zssherman) [inline act/utils/data_utils.py:1468]: Can we call new_ds_object, new_ds

**Review by zssherman — CHANGES_REQUESTED** [review]

**Comment by zssherman** [comment]
@ajsockol Thanks for the PR! I added some review comments. Also could we add unit tests to this PR as well for the new code?

**Comment by ajsockol** [comment]
I added all the changes you mentioned except the testing. I'll add that next!

**Comment by ajsockol** [comment]
Okay, I've added the tests. Now to just figure out why the linting keeps failing

**Comment by zssherman** [comment]
@ajsockol Can you try installing pre-commit? I should help with the linting issues

**Comment by mgrover1** [comment]
> @ajsockol Can you try installing pre-commit? It should help with the linting issues. I'm also trying to find out why the test_arm_qc tests are failing

It should be in your `act_env` development environment

within your main ACT directory you can run

```bash
pre-commit install .
```

Then, when you commit, it will auto-lint and fix where needed.

**Comment by zssherman** [comment]
@kenkehoe I'm still trying to familiarize with QC code of ACT, but do you possibly know why this PR might be failing the ARM QC tests?

**Comment by ajsockol** [comment]
@mgrover1 when i try to run `pre-commit install .` in my ACT directory (when I'm in my act_env development environment), I get this error:

Screenshot 2024-07-17 at 2 42 43 PM

**Comment by zssherman** [comment]
@ajsockol Maybe try pre-commit install from within the act direcctory no .

**Comment by kenkehoe** [comment]
@zssherman  I know why the QC test is failing. We use a production DQR to test the functionality of the ARM DQR web-service. The DQR used was reprocessed so it is now not returned in the API call. I have a fix for this in my current PR. It's a simple fix if we want to include it in this PR to allow it to pass the tests.

In act.tests.qc.test_arm_qc.py change line 15 to:

ds = add_dqr_to_qc(ds, assessment='Reprocessed,Suspect,Incorrect')

**Comment by zssherman** [comment]
@kenkehoe Ah makes sense! Thanks for the info! We could always wait to have your PR merge in and then can rerun the tests here.

**Comment by ajsockol** [comment]
Okay I got things figured out and was able to push changes through SSH. It looks like what's failing now are the arm_qc tests

**Comment by zssherman** [comment]
@ajsockol Cool, yeah looks like the linter etc are passing. Ken's PR fixes that QC issue so we can wait for that one to get approved then once its merge rerun the tests here.

**Comment by ajsockol** [comment]
Okay, I copied and pasted a fix from Ken for one of the tests that was failing, but there is still one test failing from tests/discovery/test_asos.py::test_get_region that seems to be something out of my hands. @zssherman do you know what might be going wrong there?

**Comment by zssherman** [comment]
@ajsockol It is a connection error, i'll rerun the tests which will fix it.

**Review by AdamTheisen — APPROVED** [review]

**Review by zssherman — APPROVED** [review]

**Pull request merged**